code/Populate.php

```diff
@@ -20,130 +25,154 @@ class Populate
+     * populating SiteTree it may be worth clearing the 'SiteTree' table.
+     *
+     * @var array
+     */
+    private static $truncate_classes = [];
+
+    /**
+     * @config
+     *
+     * @var array - Tables that will be truncated
+     */
+    private static $truncate_tables = [];
+
+    /**
+     * Flag to determine if we're already run for this session (i.e to prevent
+     * parent calls invoking {@link requireRecords} twice).
+     *
+     * @var bool
+     */
+    private static $ran = false;
+
+    /**
+     * @var string[] - Used internally to not truncate multiple tables multiple times
+     */
+    private static $clearedTables = [];
+
+    /**
+     * @param bool $force - allows you to bypass the ran check to run this multiple times
+     * @return bool
+     * @throws Exception
+     */
+    public static function requireRecords($force = false): bool
+    {
+        if (self::$ran && !$force) {
+            return true;
+        }
+
+        self::$ran = true;
+
+        if (!(Director::isDev() || Director::isTest())) {
+            throw new Exception('requireRecords can only be run in development or test environments');
+        }
+
+        $factory = Injector::inst()->create('DNADesign\Populate\PopulateFactory');
+
+        foreach (self::config()->get('truncate_classes') as $class) {
+            self::deleteClass($class);
+        }
+
+        foreach (self::config()->get('truncate_tables') as $table) {
+            self::truncateTable($table);
+        }
+
+        foreach (self::config()->get('include_yaml_fixtures') as $fixtureFile) {
+            $fixture = new YamlFixture($fixtureFile);
+            $fixture->writeInto($factory);
+
+            $fixture = null;
+        }
+
+        $populate = Injector::inst()->create(Populate::class);
+        $populate->extend('onAfterPopulateRecords');
+
+        return true;
+    }
+
+    /*
+     * Delete all the associated tables for a class
+     */
+    private static function deleteClass(string $class): void
+    {
+        // First delete the base classes
+        $tableClasses = ClassInfo::ancestry($class, true);
+        foreach ($tableClasses as $tableClass) {
+            $table = DataObject::getSchema()->tableName($tableClass);
+            self::truncateTable($table);
+        }
+
+        /** @var DataObject|FluentExtension|Versioned $obj */
+        $obj = Injector::inst()->get($class);
+
+        $versionedTables = [];
+        $hasVersionedExtension = $obj->hasExtension(Versioned::class);
+
+        if ($hasVersionedExtension) {
+            $baseTableName = Config::inst()->get($class, 'table_name');
+            $stages = $obj->getVersionedStages();
+
+            foreach ($stages as $stage) {
+                $table = $obj->stageTable($baseTableName, $stage);
+                self::truncateTable($table);
+                $versionedTables[] = $table;
+            }
+        }
+
+        if ($obj->hasExtension(FluentExtension::class)) {
```

Comment:
Instead of baking this directly into the logic, could you create an extension point here, and use the config to apply an extension for fluent, when fluent is in the project?

i.e

```
Only:
  moduleexists: 'Fluent'
```

While it's a complete breakdown if fluent is installed, it steels feels quite opinionated to include this concern here.